network/manager.go

```diff
@@ -253,3 +260,19 @@ func (mgr *Manager) AddPeer(conn net.Conn, nonce []byte) error {
 
 	return nil
 }
+
+type TcpListener struct {
```

Comment:
It should not be necessary to wrap the `net.TCPListener` into an additional struct. You can just return the `net.TCPListener` directly, since it already fulfills the interface.

Reply:
Please see comment above. If I find how to setup testify to ignore passed arguments, then we can do it.

Reply:
Actually, I think I found how to avoid it.